staging/src/k8s.io/apimachinery/pkg/runtime/serializer/allocator.go

```diff
@@ -0,0 +1,42 @@
+package serializer
+
+import (
+	"sync"
+
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// AllocatorPool simply stores Allocator objects to avoid additional memory allocations
+// by caching created but unused items for later reuse, relieving pressure on the garbage collector.
+//
+// Usage:
+//  memoryAllocator := serializer.AllocatorPool.Get().(*serializer.Allocator)
+//  defer serializer.AllocatorPool.Put(memoryAllocator)
+var AllocatorPool = sync.Pool{
+	New: func() interface{} {
+		return &Allocator{}
+	},
+}
+
+// Allocator knows how to allocate memory
+type Allocator struct {
+	buf []byte
+}
+
+var _ runtime.MemoryAllocator = &Allocator{}
+
+// Allocate reserves memory for n bytes only if the underlying array doesn't have enough capacity
+// otherwise it returns previously allocated block of memory.
+//
+// Note that the returned array is not zeroed, it is the caller's
+// responsibility to clean the memory if needed.
```

Comment:
That should go to an interface definition.

Reply:
actually, it might depend on implementation, wdyt?

Reply:
I would at least add at the interface level that implementation aren't obliged to zero the returned memory.